File: buyandsell.py
# time complexity  O(n)
# space complexity  O(1)
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if prices==[] or len(prices)==0: return
        profit=0
        for i in range(1,len(prices)):
            if prices[i-1]<prices[i]:
                profit = profit+ prices[i]-prices[i-1];
       
        return profit
    

File: buyAndSell.java
// Time complexity - O(n)
// Space complexity - O(1) - constant
// works on leetcode

class Solution {
    public int maxProfit(int[] prices) {
        if(prices == null || prices.length == 0 || prices.length == 1) {
            return 0;
        }
        int total = 0;
        for(int i=1;i<prices.length;i++){
            if(prices[i-1] < prices[i]){
                total = total + (prices[i] - prices[i-1]);
            }
        }
        return total;
    }
}
